APCAppCore/APCAppCore/Startup/APCAppDelegate.m

```diff
@@ -121,6 +123,37 @@ - (BOOL)               application: (UIApplication *) __unused application
     return YES;
 }
 
+- (void)performDemographicUploadIfRequired
+{
+    NSUserDefaults  *defaults = [NSUserDefaults standardUserDefaults];
+    
+        //
+        //    the Boolean will be NO if:
+        //        the user defaults value was never set
+        //        the actual value is NO (which should never happen)
+        //    in which case, we upload the (non-identifiable) Demographic data
+        //    Age, Sex, Height, Weight, Sleep Time, Wake Time, et al
+        //
+        //    otherwise, the value should have been set to YES, to
+        //    indicate that the Demographic data was previously uploaded
+        //
+    
+        //
+        //    we run this code iff the user has previously consented,
+        //    indicating that this is an update to a previously installed version of the application
+        //
+    APCUser  *user = ((APCAppDelegate *)[UIApplication sharedApplication].delegate).dataSubstrate.currentUser;
+    if (user.isConsented) {
+        BOOL  demographicDataWasUploaded = [defaults boolForKey:kDemographicDataWasUploadedKey];
+        if (demographicDataWasUploaded == NO) {
+            self.demographicUploader = [[APCDemographicUploader alloc] init];
+            [defaults setBool:YES forKey:kDemographicDataWasUploadedKey];
+            [defaults synchronize];
+            [self.demographicUploader uploadNonIdentifiableDemographicData];
```

Comment:
Does “-uploadNonIdentifiableDemographicData” block? If so, that should be moved to a background queue.